pandas/tests/io/parser/compression.py

```diff
@@ -166,6 +166,21 @@ def test_read_csv_infer_compression(self):
 
         inputs[3].close()
 
+    def test_read_csv_infer_compression_pathlib(self):
+        """
+        Test that compression is inferred from pathlib.Path paths.
+        """
+        try:
```

Comment:
As of https://github.com/pandas-dev/pandas/pull/17206/commits/a3b9ee9b558588bf2973aa90c5d12602799c1bb1, this test should fail when `pathlib` is available and be skipped otherwise.

Once I commit the enhancement, this test should pass when  `pathlib` is available and be skipped otherwise.